# Assignment 3.2 (R programming)

### In this assignment, you will be using data on the Los Angeles Dodgers Major League Baseball (MLB) team located here: dodgers.csv. 
### Use this data to make a recommendation to management on how to improve attendance. Tell a story with your analysis and clearly explain the steps you take to arrive at your conclusion. 
### This is an open-ended question, and there is no one right answer. You are welcome to do additional research and/or use domain knowledge to assist your analysis, but clearly state any assumptions you make.
### You can use R or Python to complete this assignment. Submit your code and output to the submission link. Make sure to add comments to all of your code and to document your steps, process, and analysis.

In [2]:
# install packages
install.packages("MASS")
install.packages("ggplot2")

Installing package into 'C:/Users/kadams/OneDrive - Suncor Energy Inc/Documents 1/R/win-library/3.6'
(as 'lib' is unspecified)



  There is a binary version available but the source version is later:
     binary source needs_compilation
MASS 7.3-54 7.3-56              TRUE

  Binaries will be installed
package 'MASS' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\kadams\AppData\Local\Temp\Rtmp4AlWbc\downloaded_packages


Installing package into 'C:/Users/kadams/OneDrive - Suncor Energy Inc/Documents 1/R/win-library/3.6'
(as 'lib' is unspecified)
also installing the dependencies 'lattice', 'nlme', 'Matrix', 'mgcv'




  There are binary versions available but the source versions are later:
         binary  source needs_compilation
lattice 0.20-44 0.20-45              TRUE
nlme    3.1-152 3.1-157              TRUE
Matrix    1.3-3   1.4-1              TRUE
mgcv     1.8-35  1.8-40              TRUE
ggplot2   3.3.3   3.3.5             FALSE

  Binaries will be installed
package 'lattice' successfully unpacked and MD5 sums checked
package 'nlme' successfully unpacked and MD5 sums checked
package 'Matrix' successfully unpacked and MD5 sums checked
package 'mgcv' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\kadams\AppData\Local\Temp\Rtmp4AlWbc\downloaded_packages


installing the source package 'ggplot2'



## Approach: The Akaike information criterion (AIC) is a metric that is used to compare the fit of different regression models. It is used to mathmatically evaluate how well the model fits the data. Lowest AIC is the best model. 
## R will be used to evaluate AIC as R provides a summary of all AIC values in 1 output, where Python does not have this summary (or at least not one that I could find online). In Python you have to input all feature combinations manually and compare the AIC values to each other. 

In [10]:
# set seed to random number, 42
set.seed(42)

In [11]:
# import csv file to variable dodgers
dodgers<-read.csv("dodgers-2022.csv", header=TRUE)
# create linear resression model using dodgers variable with attend as target variable against all features
dodgers.full<-lm(attend~., data=dodgers)
# use the MASS library
library(MASS)
# perform stepwise model selection by AIC
stepAIC(dodgers.full)

Start:  AIC=1433.33
attend ~ month + day + day_of_week + opponent + temp + skies + 
    day_night + cap + shirt + fireworks + bobblehead

              Df Sum of Sq        RSS    AIC
- opponent    16 482404341 2055399107 1423.0
- month        6 107253092 1680247858 1426.7
- temp         1     48765 1573043531 1431.3
- skies        1     73327 1573068093 1431.3
- shirt        1   1648173 1574642938 1431.4
- day          1  31985882 1604980647 1433.0
- day_night    1  34516507 1607511273 1433.1
<none>                     1572994766 1433.3
- cap          1  44252037 1617246803 1433.6
- fireworks    1 208621672 1781616438 1441.4
- bobblehead   1 305478664 1878473429 1445.7
- day_of_week  6 622226954 2195221720 1448.3

Step:  AIC=1422.99
attend ~ month + day + day_of_week + temp + skies + day_night + 
    cap + shirt + fireworks + bobblehead

              Df Sum of Sq        RSS    AIC
- day_night    1   1939503 2057338610 1421.1
- skies        1   3025454 2058424560 1421.1
- temp         


Call:
lm(formula = attend ~ month + day_of_week + cap + shirt + fireworks + 
    bobblehead, data = dodgers)

Coefficients:
         (Intercept)              monthAUG              monthJUL  
             22232.8                3341.1                2564.3  
            monthJUN              monthMAY              monthOCT  
              6933.4               -1924.8                 221.8  
            monthSEP     day_of_weekMonday   day_of_weekSaturday  
              -102.9               10931.4               17884.0  
   day_of_weekSunday   day_of_weekThursday    day_of_weekTuesday  
             18225.2               12245.2               19352.0  
day_of_weekWednesday                capYES              shirtYES  
             12463.7               -5784.6                5964.5  
        fireworksYES         bobbleheadYES  
             16309.2               10700.1  


### Result: Lowest AIC=1417.21. Attendance is best explained by the features cap, shirt, fireworks, month, day of the week, and bobblehead. 

In [12]:
# create a linear model using the features from the lowest AIC value: cap, shirt, fireworks, month, day of the week, and bobblehead. 
selected.model<-lm(formula = attend ~ month + day_of_week + cap + shirt + fireworks + bobblehead, data = dodgers)
# print the summary of the selected.model
summary(selected.model)


Call:
lm(formula = attend ~ month + day_of_week + cap + shirt + fireworks + 
    bobblehead, data = dodgers)

Residuals:
   Min     1Q Median     3Q    Max 
 -8748  -3846   -602   2435  15966 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)           22232.8     6583.3   3.377  0.00125 ** 
monthAUG               3341.1     2304.3   1.450  0.15196    
monthJUL               2564.3     2555.9   1.003  0.31951    
monthJUN               6933.4     2573.8   2.694  0.00901 ** 
monthMAY              -1924.8     2175.8  -0.885  0.37966    
monthOCT                221.8     3825.9   0.058  0.95395    
monthSEP               -102.9     2367.3  -0.043  0.96545    
day_of_weekMonday     10931.4     6802.8   1.607  0.11300    
day_of_weekSaturday   17884.0     6658.4   2.686  0.00920 ** 
day_of_weekSunday     18225.2     6712.9   2.715  0.00851 ** 
day_of_weekThursday   12245.2     6947.4   1.763  0.08275 .  
day_of_weekTuesday    19352.0     6811.9   2.84

### Results: The Estimate values measure how the features impact attendance. For example, games in the month of August increase attendance by 3341, where games in the month of May decrease by 1924. Pr values are probability, and anything under 0.05 is a good indicator that the estimate value is accurate. 
### Findings: These features all have Pr scores under 0.05 and are therefore their estimates are most likely fairly accurate:
$\bullet$ Games in the month of June<br>
$\bullet$ Game days on Saturday, Sunday, Tuesday, Wednesday<br>
$\bullet$ Games with Fireworks <br>
$\bullet$ Games with free bobbleheads<br>
### Findings in Comparison to Correlation Matrix: The correlation matrix found that attendance is highly postively correlated to: 
$\bullet$ The months of June, July & August <br>
$\bullet$ Tuesday & Saturday games <br>
$\bullet$ Games against the Angels, Cubs, Mets, Nationals & White Sox <br>
$\bullet$ Games on clear sky days <br>
$\bullet$ Game days when free shirts are given out <br>
### A highly negatively correlated to: 
$\bullet$ the months of April, May, September & October<br>
$\bullet$ Wednesday & Monday games<br>
$\bullet$ games against the Astros, Braves, Bruins, Pirates, Rockies & Snakes<br>
$\bullet$ games on cloudy days<br>
$\bullet$ game days when no free shirts are given out. <br>
### The differences between the correlated features and features with a low Pr score are related to an age old term: Correlation is not causation. Just because something is correlated doesn't mean it is statistically significant.

### Recommendations for the Dodgers to increase attendance using AIC instead of the Correlation Matrix: 
$\bullet$ Games played in June tend to increase attendance. The recommendation is for the Dodgers to schedule more games in this month.  <br>
$\bullet$ Games played on Saturdays, Sundays, and Tuesdays tend to increase attendance, with Tuesdays with the most attendance increase estimated. The recommendation is for the Dodgers to schedule more games on these days. <br>
$\bullet$ Games with a fireworks show and games where free bobbleheads are given out tend to increase attendance, with fireworks having the most attendance increase estimated. The recommendation is for the Dodgers to include more events on game days. <br>